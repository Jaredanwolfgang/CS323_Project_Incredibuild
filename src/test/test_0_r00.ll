source_filename = "test_0_r00.spl"

define i32 @main() #0 {
  %1 = alloca i32, align 4
  %2 = alloca i32, align 4
  store i32 0, ptr %1, align 4
  store i32 3, ptr %2, align 4
  br label %3

3:                                                ; preds = %0, %9
  %4 = load i32, ptr %2, align 4
  %5 = add nsw i32 %4, 1
  store i32 %5, ptr %2, align 4
  %6 = load i32, ptr %2, align 4
  %7 = icmp eq i32 %6, 5
  br i1 %7, label %8, label %9

8:                                                ; preds = %3
  br label %10

9:                                                ; preds = %3
  br label %3

10:                                               ; preds = %8
  %11 = load i32, ptr %2, align 4
  ret i32 %11
}