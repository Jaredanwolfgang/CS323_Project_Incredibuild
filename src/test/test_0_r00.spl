int main(){
    int a = 3;
    while (true){
        a = a + 1;
        if (a == 5){
            break;
        }
    }
    return a;
}
