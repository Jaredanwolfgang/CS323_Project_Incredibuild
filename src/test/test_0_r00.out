Function: main:[Body: [Variable Declaration: a = [0: i32] with dimensions []; Variable Assignment: a = 3: i32, While Loop (Condition: true): 
 do Body: [Variable Assignment: a = (a + 1: i32), If: Condition: a == 5: i32 then Body: [Break]], Return: null]]
